compute/cker/src/DepthwiseConv.test.cc

```diff
@@ -0,0 +1,320 @@
+        {batch, filter_spatial_size, padded_filter_inner_dim_size});
+      _padded_filter.resize(padded_filter_shape.FlatSize());
+    }
+
+    {
+      // NOTE The Eigen library uses both main thread as well as a thread pool.
+      // Therefore, it needs to add an additional memory buffer for main thread.
+      const int thread_count = nnfw::cker::eigen_support::getThreadCount() + 1;
+
+      nnfw::cker::Shape filter_buffer_shape(
+        {thread_count, filter_spatial_size, padded_filter_inner_dim_size});
+      _filter_buffers.resize(filter_buffer_shape.FlatSize());
+    }
+  }
+
+  void run(const nnfw::cker::DepthwiseConvParams &params, const nnfw::cker::Shape &input_shape,
+           const T *input_data, const nnfw::cker::Shape &filter_shape, const T *filter_data,
+           const nnfw::cker::Shape &bias_shape, const T *bias_data,
+           const nnfw::cker::Shape &output_shape, const T *expected)
+  {
+    std::vector<T> output(output_shape.FlatSize());
+    nnfw::cker::DepthwiseConvOp(params, input_shape, input_data, filter_shape, filter_data,
+                                bias_shape, bias_data, _padded_filter.data(), _use_padded_filter,
+                                _filter_buffers.data(), output_shape, output.data());
+
+    for (size_t i = 0; i < output.size(); ++i)
+      EXPECT_NEAR(output[i], expected[i], 1e-3f);
+  }
+
+  void checkException(const nnfw::cker::DepthwiseConvParams &params,
+                      const nnfw::cker::Shape &input_shape, const T *input_data,
+                      const nnfw::cker::Shape &filter_shape, const T *filter_data,
+                      const nnfw::cker::Shape &bias_shape, const T *bias_data,
+                      const nnfw::cker::Shape &output_shape, const T *expected)
+  {
+    std::vector<T> output(output_shape.FlatSize());
+    EXPECT_ANY_THROW(
+      nnfw::cker::DepthwiseConvOp(params, input_shape, input_data, filter_shape, filter_data,
+                                  bias_shape, bias_data, _padded_filter.data(), _use_padded_filter,
+                                  _filter_buffers.data(), output_shape, output.data()));
+  }
+
+  void checkDeath(const nnfw::cker::DepthwiseConvParams &params,
+                  const nnfw::cker::Shape &input_shape, const T *input_data,
+                  const nnfw::cker::Shape &filter_shape, const T *filter_data,
+                  const nnfw::cker::Shape &bias_shape, const T *bias_data,
+                  const nnfw::cker::Shape &output_shape, const T *expected)
+  {
+    std::vector<T> output(output_shape.FlatSize());
+    EXPECT_DEATH(nnfw::cker::DepthwiseConvOp(params, input_shape, input_data, filter_shape,
+                                             filter_data, bias_shape, bias_data,
+                                             _padded_filter.data(), _use_padded_filter,
+                                             _filter_buffers.data(), output_shape, output.data()),
+                 "");
```

Comment:
This regular expression(or matcher) probably won't fail at all. That means this test is always passed even if the condition of the corresponding assertion is not `true`.

Reply:
As we saw offline, `""` means to allow all error messages, not to prevent the test from failing. The reason this test passed in release mode was because the test died for another reason(maybe seg fault), not the assertion.